packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -37,22 +37,21 @@ import {
   getFirstFocusableDescendant,
 } from '../../utils/accessibility-utils';
 
-export interface BaseResultList<T = FoldedResult | Result>
-  extends InitializableComponent {
+export interface BaseResultList extends InitializableComponent {
   host: HTMLElement;
   templateHasError: boolean;
   resultListCommon: ResultListCommon;
 
-  setRenderFunction?: SetRenderFunction<T>;
+  setRenderFunction?: SetRenderFunction;
```

Comment:
the idea behind doing this was removing the need to check what kind of result it is in the rendering function, for example `function MyTemplate(result: FoldedResult)` works when passed to `AtomicFoldedResultList` and `function MyTemplate(result: Result) {` works when passed to `AtomicResultList`.
is the user now going to have to do 
```ts
function MyTemplate(result: Result | FoldedResult) {
  if (isFolded(result)) { etc... }
```
or is there some other magic that does this that I didn't see?

Reply:
The cast happens internally in `ResultListWrapper` and `FoldedResultListWrapper` in Atomic React, which is code we own.

For implementers using Atomic React, they are still receiving the correct type in their templates, so it's transparent and they don't have to do any cast.

By removing the generic here, it's simpler to reuse the `ResultRenderingFunction` type across more files.